### PR TITLE
ethereum as generic upstream

### PR DESCRIPTION
**Pull request by a10zn8**

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/config/CacheConfig.kt [hunk 1]
```diff
@@ -17,8 +17,6 @@ package io.emeraldpay.dshackle.config
 
 class CacheConfig {
 
-    var requestsCacheEnabled = true
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/config/CacheConfig.kt:20]: Do we need to remove this param? Does it mean that caches are always enabled?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt [hunk 2]
```diff
@@ -1,14 +1,87 @@
...
+        return Mono.just(EthereumLocalReader(cachingReader as EthereumCachingReader, methods, head))
+    }
+
+    override fun subscriptionBuilder(headScheduler: Scheduler): (Multistream) -> EgressSubscription {
+        return { ms ->
+//            val pendingTxes: PendingTxesSource = (ms.getAll())
+//            .mapNotNull {
+//                it.getIngressSubscription().getPendingTxes()
+//            }.let {
+//                if (it.isEmpty()) {
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt:48]: remove it?

**Review by KirillPamPam — APPROVED** [review]

File: src/main/kotlin/io/emeraldpay/dshackle/config/CacheConfig.kt (hunk 1 above)
Comment (a10zn8) [reply src/main/kotlin/io/emeraldpay/dshackle/config/CacheConfig.kt:20]: yes. It was a hack for dshackle main, now it useless

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt (hunk 2 above)
Comment (a10zn8) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumChainSpecific.kt:48]: Will fix it, last one

**Pull request merged**